python/test/gluon/test_core.py

```diff
@@ -106,6 +107,61 @@ def test_tma():
     torch.testing.assert_close(out, torch.zeros_like(out))
 
 
+@gluon.jit
+def tma_multicast_copy_kernel(in_desc, out_desc):
+    smem = ttgl.allocate_shared_memory(in_desc.dtype, in_desc.block_shape, in_desc.layout)
+
+    bar = mbarrier.allocate_mbarrier()
+    mbarrier.init(bar, count=1)
+    # Need to synchronise all the CTAs after the mbarrier initialisation
+    # so that they all see it before tma.async_copy_global_to_shared(multicast=True)
+    cluster_sync()
+
+    mbarrier.expect(bar, in_desc.bytes_per_cta)
```

Comment:
I see, this isn't what we discussed though. You suggested it would be
```suggestion
    mbarrier.expect(bar, in_desc.nbytes)
```
And the expect op would internally distribute the nbytes according to the num_cta_split.

Reply:
Ah, because you can have broadcasting in the data but still have separate barriers. That's unfortunate.

Perhaps it would be better to have:
```python
    mbarrier.expect(bar, descs=(a_desc, b_desc))
```
And hide the calculations from the user.

Reply:
ugh, I don't like that API really. I'd prefer to just expose `descs` to keep the API simple.

Reply:
Changing the API would be an absolute pain, so I'm exposing both under the same function, but I will not allow to use them both. WDYT. Do you prefer this or do you prefer a different `def expect_descs(descs)`?